vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -182,6 +182,13 @@ def process_weights_after_loading(self, layer: Module) -> None:
         # If checkpoint is fp8, handle that there are N scales for N
         # shards in a fused module
         else:
+            # If rocm, use float8_e4m3fnuz
+            if isinstance(current_platform, rocm.RocmPlatform):
+                weight_as_int8 = layer.weight.view(torch.int8)
+                weight_as_int8[weight_as_int8 == -128] = 0
```

Comment:
We do not expect `NaN` in weights for inference, so flag an exception vs. reset to `0`.

Reply:
0b10000000 is -0 in OCP, but is a NaN on NANOO, it needs this special handling

Reply:
Could you pull out `-128` as a named constant then? `ROCM_FP8_NAN = -128` or whatever is accurate here